libsql-sqlite3/src/pager.c

```diff
@@ -697,10 +699,8 @@ struct Pager {
   char *pTmpSpace;            /* Pager.pageSize bytes of space for tmp use */
   PCache *pPCache;            /* Pointer to page cache object */
 #ifndef SQLITE_OMIT_WAL
-  libsql_wal *pWal;                       /* Write-ahead log used by "journal_mode=wal" */
-  libsql_wal_methods *pWalMethods; /* Virtual methods for interacting with WAL */
-  char *zWal;                      /* File name for write-ahead log */
-  void *pWalMethodsData;           /* WA: methods context data */
+  RefCountedWalManager* wal_manager;
+  libsql_wal wal;
```

Comment:
let's not blow the pager so much and make its sizeof dependent on `sizeof(libsql_wal)`, `wal` should remain a pointer.